src/engine.rs

```diff
@@ -648,3 +652,42 @@ impl<'a> DownloadsMapped<'a> {
         Ok(slice.get_mapped_range())
     }
 }
+
+const SIZE_CLASS_BITS: u32 = 1;
+
+impl ResourcePool {
+    /// Get a buffer from the pool or create one.
+    fn get_buf(&mut self, size: u64, usage: BufferUsages, device: &Device) -> Buffer {
```

Comment:
As the scene grows across frames, this currently becomes unbounded. We should have a garbage collection scheme of some time (e.g. a time last used counter on each buffer, clearing when they're more than 3 seconds old or something)

Reply:
If you do the math, in the case of a scene growing, the actual total memory allocation is a reasonably small constant multiple of the working set. The constant depends on the number of bits used for size_class. The story also gets better if there's one big buffer and a bunch of smaller buffers that can reuse the allocations left over from the big ones.

That said, yes we should have eviction, but the heuristic to do it *well* is tricky, and I don't think it's blocking anything that we're doing right now. If we wanted to land this PR, I would feel comfortable having an issue.

Reply:
I think, as we do currently, using the front two bits, the 'lost' buffer is 75% the size of the current[^1]. That means the cost is at most a factor of:

$$\frac{1}{1 - 0.75} = 4$$

That's not that huge of a cost to eat, I agree, when our biggest supported buffer is on the order of 16MB (I think, anyway). It's certainly not ideal though. I'm happy landing this as-is, at least for now.

[^1]: Half the time, it's actually 66% the size, but I'm ignoring that because it makes my calculation easier. Also ignoring the case of size 0 and 1 as negligible